### PR TITLE
Custom mapping relationships

### PR DESCRIPTION
**Pull request by RamonGilabert**
:dancer: I guess we could change the name to add unity between `Sync` and `PropertyMapper`, the name of the key is `sync.remote.key` now, however, for the user, it would be easier if the keys are the same: `mapper.remote.key`.

Also, it's only done for One to Many relations at the moment, let me know! :smile:

### CODE REVIEW

File: Source/Sync.m [hunk 1]
```diff
@@ -188,10 +188,21 @@ - (void)sync_processToManyRelationship:(NSRelationshipDescription *)relationship
                              andParent:(NSManagedObject *)parent
                              dataStack:(DATAStack *)dataStack
 {
+    NSString *relationshipKey = [[relationship userInfo] valueForKey:SyncCustomRemoteKey];
+
+    BOOL hasCustomMapper = (relationshipKey);
+
+    NSString *relationshipName;
```
Comment (3lvis) [inline Source/Sync.m]: Here you could do 

```objc
NSString *relationshipName = (hasCustomMapper) ? relationshipKey : relationship.name;
```

File: Tests/Tests/Tests.m [hunk 2]
```diff
@@ -369,4 +369,23 @@ - (void)testCustomPrimaryKeyInsideToManyRelationship
...
 
+- (void)testCustomKeysInRelationships
+{
+    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+    NSArray *objects = [NSJSONSerialization JSONObjectWithContentsOfFile:@"custom_relationship_key_JSON.json"
+                                                                inBundle:bundle];
+    [Sync processChanges:objects
+         usingEntityName:@"User"
+               dataStack:self.dataStack
+              completion:^(NSError *error) {
```
Comment (3lvis) [inline Tests/Tests/Tests.m]: I like how now you add tests for everything :heart:

File: Tests/Tests/custom_relationship_key_JSON.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[
+    {
+        "id": 10,
+        "name": "Legen Dary",
+        "email": "melisawhite@ovium.com",
+        "created_at": "2014-02-14T00:00:00+00:00",
+        "updated_at": "2014-02-17T00:00:00+00:00",
```
Comment (3lvis) [inline Tests/Tests/custom_relationship_key_JSON.json]: Check http://jsonformatter.curiousconcept.com/ for a JSON prettifier

```json
[
  {
    "id":10,
    "name":"Legen Dary",
    "email":"melisawhite@ovium.com",
    "created_at":"2014-02-14T00:00:00+00:00",
    "updated_at":"2014-02-17T00:00:00+00:00",
    "annotations":[
      {
        "id":30
      },
      {
        "id":50
      },
      {
        "id":70
      }
    ]
  }
]
```

**Comment by RamonGilabert** [comment]
Nice! :smile: now, what do you think about changing the name from: `sync.remote.key` to `mapper.remote.key`?

File: Tests/Tests/Tests.m (hunk 2 above)
Comment (RamonGilabert) [inline Tests/Tests/Tests.m]: I love them! :)

**Comment by 3lvis** [comment]
I wonder if `hyper.remote.key` would be better, it would make it consistent at least

**Comment by RamonGilabert** [comment]
Right, but then I would change it too in `PropertyMapper`

**Pull request merged**